STYLE.md

```diff
@@ -0,0 +1,446 @@
+
+```haskell
+data Tx = UnsafeTx
+  { txInputs     :: !(NonEmpty TxIn)
+  -- ^ Inputs of transaction.
+  , txOutputs    :: !(NonEmpty TxOut)
+  -- ^ Outputs of transaction.
+  , txAttributes :: !TxAttributes
+  -- ^ Attributes of transaction
+  }
+```
+
+
+Convention for field names for record data types:
+
+Choose a reasonable prefix is used such that we can ensure the field name will remain unique throughout the package. Use your discretion.
+
+e.g
+```haskell
+data MempoolEnv m blk = MempoolEnv
+  { meEnvLedger    :: LedgerInterface m blk
+  , meEnvLedgerCfg :: LedgerConfig blk
+  , meEnvStateVar  :: StrictTVar m (InternalState blk)
+  , meEnvTracer    :: Tracer m (TraceEventMempool blk)
+  }
+```
+
+### Comments
+
+#### Punctuation
+
+Write proper sentences; start with a capital letter and use proper
+punctuation.
+
+#### Top-Level Definitions
+
+* Comment top level functions (particularly exported functions)
+as much as possible. If it is not blindingly obvious what a function
+is doing, you should add a comment.
+* Always provide a type signature
+* Use Haddock syntax in the comments.
+* Comment every exported data type.
+
+Function example:
+
+```haskell
+-- | Send a message on a socket. The socket must be in a connected
+-- state. Returns the number of bytes sent. Applications are
+-- responsible for ensuring that all data has been sent.
+send
+  :: Socket      -- ^ Connected socket
+  -> ByteString  -- ^ Data to send
+  -> IO Int      -- ^ Bytes sent
+```
+
+* For functions, the documentation should give enough information to
+apply the function without looking at its definition.
+
+Record example:
+
+```haskell
+-- | Bla bla bla.
+data Person = Person
+  { age  :: !Int     -- ^ Age
+  , name :: !String  -- ^ First name
+  }
+```
+
+For fields that require longer comments, format them this way:
+
+```haskell
+data Record = Record
+  { -- | This is a very very very long comment that is split over
+    -- multiple lines.
+    field1 :: !Text
+
+    -- | This is a second very very very long comment that is split
+    -- over multiple lines.
+  , field2 :: !Int
+  }
+```
+
+#### End-of-Line Comments
+
+Separate end-of-line comments from the code with 2 spaces. Align
+comments for data type definitions. Some examples:
+
+```haskell
+data Parser =
+  Parser
+    !Int         -- ^ Current position
+    !ByteString  -- ^ Remaining input
+
+foo :: Int -> Int
+foo n = salt * 32 + 9
+  where
+    salt = 453645243  -- Magic hash salt.
+```
+
+#### Links
+
+Use in-line links economically.  You are encouraged to add links for
+API names. It is not necessary to add links for all API names in a
+Haddock comment. We therefore recommend adding a link to an API name
+if:
+
+* The user might actually want to click on it for more information (in
+  your opinion), and
+
+* Only for the first occurrence of each API name in the comment (don't
+  bother repeating a link)
+
+Top-down guideline
+------------------
+
+### LANGUAGE extensions section
```

Comment:
Maybe we should also have a section for `OPTIONS_GHC` pragma

Reply:
@intricate  Feel free to suggest them :)

Reply:
I would note where `OPTIONS_GHC` pragmas should be located in the module (in relation to `LANGUAGE` pragmas), how they should be aligned, etc.

And yeah, at the bottom of the document we could also list GHC options to avoid.